src/protagonist/Engine/Ingest/Image/Appetiser/AppetiserClient.cs

```diff
@@ -102,10 +113,19 @@ public async Task<bool> ProcessImage(IngestionContext context)
         }
 
         using var response = await httpClient.SendAsync(request);
-        response.EnsureSuccessStatusCode();
-
+                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     
         // TODO - it's possible to get a 200 when appetiser doesn't do anything, e.g. body not understood
-        var responseModel = await response.Content.ReadFromJsonAsync<AppetiserResponseModel>();
+        AppetiserResponse? responseModel = null;
+        
+        if (response.IsSuccessStatusCode)

```

Comment:
Is this guaranteed to be an indicator of the shape of message returned (thinking of the `// TODO - it's possible to get a 200 when appetiser doesn't do anything, e.g. body not understood` comment)? And if it is guaranteed to work does it depend on a specific version of Appetiser being available? 

Would it be better to look at the contents of the response message (and avoid a lock step deployment)

Reply:
I could do a check on the image id, and if it get's null, try deserializing to the error format?

For the specific version of appetiser, I need to do some testing, but I think this will work with currently deployed versions of appetiser.  However, the error would be essentially junk (it would always say `500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.`)